GetServicePreferences/handler.ts

```diff
@@ -0,0 +1,183 @@
+import { NonNegativeInteger } from "@pagopa/io-functions-commons/node_modules/@pagopa/ts-commons/lib/numbers";
+import { NonEmptyString } from "@pagopa/io-functions-commons/node_modules/@pagopa/ts-commons/lib/strings";
+import { identity } from "fp-ts/lib/function";
+
+type IGetServicePreferencesHandlerResult =
+  | IResponseSuccessJson<ServicePreference>
+  | IResponseErrorNotFound
+  | IResponseErrorConflict
+  | IResponseErrorQuery;
+
+/**
+ * Type of a GetServicePreferences handler.
+ *
+ * GetServicePreferences expects a FiscalCode as input and returns a Profile or
+ * a Not Found error.
+ */
+type IGetServicePreferencesHandler = (
+  fiscalCode: FiscalCode,
+  serviceId: ServiceId
+) => Promise<IGetServicePreferencesHandlerResult>;
+
+/**
+ *
+ * @param maybeProfile
+ * @returns
+ */
+const getProfileOrErrorResponse = (
+  maybeProfile: o.Option<RetrievedProfile>
+): te.TaskEither<IResponseErrorNotFound, Profile> =>
+  te.fromEither(
+    e.fromOption(
+      ResponseErrorNotFound(
+        "Profile not found",
+        "The profile you requested was not found in the system."
+      )
+    )(maybeProfile)
+  );
+
+/**
+ * Return a function that returns the service preference for the
+ * given documentId and version, or a default value if not present
+ * The default value depends on the user' settings (mode AUTO/MANUAL)
+ *
+ * @param servicePreferencesModel The service preferences cosmos model
+ * @param fiscalCode the fiscal code
+ * @returns
+ */
+export declare type getUserServicePreferencesT = (params: {
+  readonly documentId: NonEmptyString;
+  readonly mode:
+    | ServicesPreferencesModeEnum.AUTO
+    | ServicesPreferencesModeEnum.MANUAL;
+  readonly version: NonNegativeInteger;
+  readonly fiscalCode: FiscalCode;
+}) => te.TaskEither<IResponseErrorQuery, ServicePreference>;
+const getUserServicePreferencesOrDefault = (
+  servicePreferencesModel: ServicesPreferencesModel
+): getUserServicePreferencesT => ({ fiscalCode, documentId, mode, version }) =>
+  servicePreferencesModel
+    .find([documentId, fiscalCode])
+    .mapLeft(failure =>
+      ResponseErrorQuery(
+        "Error while retrieving the user's service preferences",
+        failure
+      )
+    )
+    .map(maybeServicePref =>
+      maybeServicePref.foldL<ServicePreference>(
+        () => {
+          switch (mode) {
+            case ServicesPreferencesModeEnum.AUTO:
+              return toDefaultEnabledUserServicePreference(version);
+            case ServicesPreferencesModeEnum.MANUAL:
+              return toDefaultDisabledUserServicePreference(version);
+          }
+        },
+        pref => toUserServicePreferenceFromModel(pref)
+      )
+    );
+
+/**
+ * Return a type safe GetServicePreferences handler.
+ */
+export const GetServicePreferencesHandler = (
+  profileModels: ProfileModel,
+  servicePreferencesModel: ServicesPreferencesModel
+): IGetServicePreferencesHandler => {
+  return async (fiscalCode, serviceId) => {
+    const p = profileModels
```

Comment:
just return it ;)

Reply:
done in [42bd076](https://github.com/pagopa/io-functions-app/pull/170#:~:text=just%20return%20it-,42bd076,-Merge%20state)